docs/builtins.rst

```diff
@@ -0,0 +1,217 @@
+
+      first-user = Primarely, { TAKE($users, 1) } likesd your message.
+
+``DROP``
+    Returns a slice of the list starting from a given index.
+    Similar to JS `Array.prototype.slice`_ API.
+
+    Example::
+
+      more-users = But ultimately, { DROP($users, 1) } like your message as well.
+
+
+Implicit use
+============
+
+In order to simplify most common scenarios, L20n will run a default
+built-in while resolving placeables.
+
+For the list of implicit built-ins, the implict example has exactly the same
+result as the explicit one.
+
+``NUMBER``
+    If the variable passed from the developer is a number and is used in
+    a placeable, L20n will implicitly call a `NUMBER` built-in on it.
+
+    Example::
+
+      emails = Number of unread emails { $unreadEmails }
+
+      emails2 = Number of unread emails { NUMBER($undeadEmails) }
+
+``DATETIME``
+    If the variable passed from the developer is a date and is used in
+    a placeable, L20n will implicitly call a `DATE` built-in on it.
+
+    Example::
+
+      log-time = Entry time: { $date }
+
+      log-time2 = Entry time: { DATETIME($date) }
+
+``PLURAL``
+    If the variable passed from the developer is a number and is used in
+    a selector expression, L20n will implicitly call a `PLURAL` built-in on it.
+
+    Example::
+
+      liked-count = { PLURAL($num) ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+
+      liked-count2 = { $num ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+
+``LIST``
+    If the variable passed from the developer is a number and is used in
+    a placeable, L20n will implicitly call a `LIST` built-in on it.
+
+    Also, if the placeable is a list of variables, L20n will implicitly
+    call a `LIST` built-in on it.
+
+    Example::
+
+      users = { LIST($user1, $user2, $user3) }
+
+      users2 = { $user1, $user2, $user3 }
+
+      users = { LIST($users) }
+
+      users2 = { $users }
+
+
+Partially resolved built-ins
+============================
+
+In the future we'll want to allow for partially resolved builtins to be
+constructed by the developer and passed to the localization.
+This scenario will be used when the developer wants to define the
+default parameters for the built-in, but allow the localizer
+to override them if needed.
+
+It may look like this::
+
+  main.js:
+
+  let s = document.l10n.formatValue('key1', {
+    'date': L20n.DateTime({
+      month: 'long',
+      year: 'numeric',
+      day: 'numeric'
+    })
+  });
+
+  main.ftl:
+
+  # If the localizers doesn't need to change anything
+  key1 = Current date is { $date }
+
+  # If the localizer needs to change the way we display month to fit the text
+  key1 = Current date is { DATETIME($date, month=short) }
+
+
+Gecko runtime specific built-ins
+================================
+
+At the moment Gecko supports the following OS-specific built-ins:
```

Comment:
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1311666 to track the renaming.  Let's not block on it.